src/adagrams.js

```diff
@@ -1,15 +1,145 @@
+//helper functions
+
+// const randomInt = function (max) {
+//   return Math.floor(Math.random() * max);
+// };
+const randomInt = (max) => Math.floor(Math.random() * max);
+
+const removeLetter = function (arr, target) {
+  let i = arr.indexOf(target);
+  if (i > -1) {
+    arr.splice(i, 1);
+  }
+  return arr;
+};
+
+const uppercaseArray = (str) => {
+  let string = str.toUpperCase();
+  return Array.from(string);
+};
+
+//main functions
+
+//Wave 1
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const lettersObject = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+  //create bag of letters
+  const bagOfLetters = [];
+  Object.keys(lettersObject).forEach((letter) => {
+    for (let i = 0; i < lettersObject[letter]; i++) {
+      bagOfLetters.push(letter);
+    }
+  });
+
+  //draw a hand of 10 random letters
+  const hand = [];
+  while (hand.length !== 10) {
+    let letterIndex = randomInt(bagOfLetters.length);
+    hand.push(bagOfLetters[letterIndex]);
+    removeLetter(bagOfLetters, bagOfLetters[letterIndex]);
+  }
+  return hand;
 };
 
+//wave 2
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const inputArray = uppercaseArray(input);
+  for (let letter of inputArray) {
+    if (lettersInHand.includes(letter) === false) {
+      return false;
+    }
+    removeLetter(lettersInHand, letter);
+  }
+  return true;
 };
 
+//wave 3
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  let score = 0;
+  if (word === false) {
+    return 0;
+  }
+
+  const scoreboard = {
```

Comment:
This can be declared outside of the function. Also, it would be more efficient to store each letter as the key and the points for that letter as the value. Looking up a key in an object is O(1) time complexity. It's more efficient than looking at every single key:value pair in the object to find the one that matches the specific letter you're looking for.